packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -75,6 +75,10 @@ export class StripeWebhookHandler extends StripeHandler {
             await this.handleCreditNoteEvent(request, event);
           }
           break;
+        case 'coupon.created':
+        case 'coupon.updated':
+          await this.handleCouponEvent(request, event);
+          break;
```

Comment:
I didn't see any issues that could be caused by promotion codes failing to apply, as its the coupon in the case this issue is resolving that was incorrectly configured to require a product id. We didn't talk about validation requirements on a promotion code during the creation of this issue. It wasn't clear to me whether a product id requirement could be added later so I captured updated events to err on the safe side.

Reply:
Re: promotion codes: I don't think we knew at the time (or at least I didn't), that you could trigger promotionCode webhook events from editing fields on the Coupon page. This would only help in catching misconfigured coupons that were made prior to this patch landing.

It will still be helpful to check `coupon.updated` for those coupons that have already been created that have "Applicable products".